cmd/atlas/internal/cmdapi/migrate.go

```diff
@@ -396,6 +411,123 @@ func CmdMigrateNewRun(_ *cobra.Command, args []string) error {
 	return migrate.NewPlanner(nil, dir, migrate.WithFormatter(f)).WritePlan(&migrate.Plan{Name: name})
 }
 
+// CmdMigrateStatusRun is the command executed when running the CLI with 'migrate status' args.
+func CmdMigrateStatusRun(cmd *cobra.Command, _ []string) error {
+	// Open the migration directory.
+	dir, err := dir()
+	if err != nil {
+		return err
+	}
+	sc := dir.(migrate.Scanner) // all supported migration directories implement the migrate.Scanner
+	avail, err := sc.Files()
+	if err != nil {
+		return err
+	}
+	// Open a client to the database.
+	client, err := sqlclient.Open(cmd.Context(), MigrateFlags.URL)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+	if ok, err := revisionsTableExists(cmd.Context(), client); !ok || err != nil {
+		if err != nil {
+			return err
+		}
+		return errors.New("revisions table does not exist")
+	}
+	// Currently, only in DB revisions are supported.
+	opts := []entmigrate.Option{entmigrate.WithSchema(MigrateFlags.RevisionSchema)}
+	rrw, err := entmigrate.NewEntRevisions(client, opts...)
+	if err != nil {
+		return err
+	}
+	if err := rrw.Init(cmd.Context()); err != nil {
+		return err
+	}
+	// Executor can give us insights on the revision state.
+	ex, err := migrate.NewExecutor(client.Driver, dir, rrw)
+	if err != nil {
+		return err
+	}
+	dirty := migrate.ErrDirty{}
+	pending, err := ex.Pending(context.Background())
+	if err != nil && !errors.As(err, &dirty) && !errors.Is(err, migrate.ErrNoPendingFiles) {
+		return err
+	}
+	var (
+		applied                 = avail[:len(pending)+1]
+		prev, cur, next, latest string
```

Comment:
I don't really understand what each state means. Can you elaborate, please?

Reply:
`prev`: previously applied version (like `20221010235500`)
`cur ` : currently applied version
`next`: next version in line
`latest`: latest version in the migration directory

Reply:
Well, we have a `--dry-run` flag as well even though we could simply look inside of the files. This is nice-to-have. But I don't have strong feelings about those two. The important part is to know the migration state and what currently is applied.

Reply:
Can I propose starting with `current` which is the latest applied ("current database version"), and `next` for "next migration file to apply"? Then, we can play with it a bit and improve it if it's needed.

Reply:
cc @rotemtam

Reply:
![image](https://user-images.githubusercontent.com/12862103/178250698-ed718787-2cbd-44de-872f-8b208c65ac0e.png)